<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Sample Analysis • vicar</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Sample Analysis">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">vicar</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/customFA.html">Customizable Factor Analyses</a>
    </li>
    <li>
      <a href="../articles/custom_prior.html">Custom Prior for RUVB</a>
    </li>
    <li>
      <a href="../articles/sample_analysis.html">Sample Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dcgerard/vicar">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Sample Analysis</h1>
                        <h4 class="author">David Gerard</h4>
            
            <h4 class="date">2016-12-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/dcgerard/vicar/blob/master/vignettes/sample_analysis.Rmd"><code>vignettes/sample_analysis.Rmd</code></a></small>
      <div class="hidden name"><code>sample_analysis.Rmd</code></div>

    </div>

    
    
<div id="abstract" class="section level1">
<h1 class="hasAnchor">
<a href="#abstract" class="anchor"></a>Abstract</h1>
<p>In this vignette, I take an example simulated RNA-seq dataset and analyze it using the functions in the vicar R package. I compare their performances against the methods available in other packages.</p>
</div>
<div id="data" class="section level1">
<h1 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h1>
<p>The data are simulated RNA-seq expression data based on the characteristics of the GTEx data: . The overall model for these data is <span class="math display">\[
Y = X\beta + Z\alpha + E,
\]</span> where <span class="math inline">\(Y\)</span> contains the gene expression levels, <span class="math inline">\(X\)</span> contains the observed covariates, <span class="math inline">\(\beta\)</span> contains the coefficients of the observed covariates, <span class="math inline">\(Z\)</span> contains the unobserved confounders, <span class="math inline">\(\alpha\)</span> contains the coefficients of the unobserved confounders, and <span class="math inline">\(E\)</span> contains independent Gaussian noise with column-specific variances. The data, <code>sim_gtex</code> consists of a list of elements:</p>
<ul>
<li>
<code>Y</code> A matrix of gene expression levels. The rows index the samples and the columns index the genes.</li>
<li>
<code>X</code> A matrix of covariates. The first column is simply for an intercept term and the second column is a group indicator.</li>
<li>
<code>beta</code> The known signal for the group indicator.</li>
<li>
<code>ctl</code> A logical vector indicating the location of control genes.</li>
<li>
<code>which_null</code> A logical vector indicating the location of null genes.</li>
</ul>
<p>I’ve added signal to the <code>Y</code> matrix, the amount of which is encoded in the vector <code>beta</code>. We can read in these data using the <code>data</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(vicar)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</code></pre></div>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(sim_gtex)
Y &lt;-<span class="st"> </span>sim_gtex<span class="op">$</span>Y
X &lt;-<span class="st"> </span>sim_gtex<span class="op">$</span>X
ctl &lt;-<span class="st"> </span>sim_gtex<span class="op">$</span>ctl
which_null &lt;-<span class="st"> </span>sim_gtex<span class="op">$</span>which_null
beta &lt;-<span class="st"> </span>sim_gtex<span class="op">$</span>beta</code></pre></div>
<p>These data contain 20 samples and 1000 genes. The proportion of genes in these data that are null is 0.9. However, we only allow 10 of these genes to be known as negative controls.</p>
<p>Before we proceed with the analysis, we note that all confounder adjustment methods we explore require an estimate of the number of hidden confounders. We can use the <code>num.sv</code> function in the sva package, or the <code>est.confounder.num</code> function in the cate package to obtain this estimate. They each say that there are about 3 or 4 hidden confounders.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">num_sv     &lt;-<span class="st"> </span>sva<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/sva/topics/num.sv">num.sv</a></span>(<span class="dt">dat =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(Y), <span class="dt">mod =</span> X, <span class="dt">method =</span> <span class="st">"be"</span>)
num_sv_l   &lt;-<span class="st"> </span>sva<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/sva/topics/num.sv">num.sv</a></span>(<span class="dt">dat =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(Y), <span class="dt">mod =</span> X, <span class="dt">method =</span> <span class="st">"leek"</span>)
num_sv_bcv &lt;-<span class="st"> </span>cate<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/cate/topics/est.confounder.num">est.confounder.num</a></span>(<span class="op">~</span><span class="st"> </span>Intercept <span class="op">+</span><span class="st"> </span>Treatment, <span class="dt">X.data =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(X), <span class="dt">Y =</span> Y,
                                       <span class="dt">method =</span> <span class="st">"bcv"</span>, <span class="dt">bcv.plot =</span> <span class="ot">FALSE</span>, <span class="dt">nRepeat =</span> <span class="dv">50</span>)<span class="op">$</span>r
num_sv</code></pre></div>
<pre><code>## [1] 3</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">num_sv_l</code></pre></div>
<pre><code>## [1] 5</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">num_sv_bcv</code></pre></div>
<pre><code>## [1] 3</code></pre>
<p>For this vignette, we’ll use the output from <code>num.sv</code>, which indicates that there are 3 hidden confounders.</p>
</div>
<div id="analysis-with-controls" class="section level1">
<h1 class="hasAnchor">
<a href="#analysis-with-controls" class="anchor"></a>Analysis With Controls</h1>
<p>When there are control genes, there are many options to analyze your data. vicar, cate, and ruv all have their different versions of RUV4. We’ll just look at the default settings for each.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ruv4_vicar &lt;-<span class="st"> </span>vicar<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/vicar/topics/vruv4">vruv4</a></span>(<span class="dt">Y =</span> Y, <span class="dt">X =</span> X, <span class="dt">k =</span> num_sv, <span class="dt">ctl =</span> ctl, <span class="dt">cov_of_interest =</span> <span class="dv">2</span>)
ruv4_cate  &lt;-<span class="st"> </span>cate<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/cate/topics/cate">cate.fit</a></span>(<span class="dt">X.primary =</span> X[, <span class="dv">2</span>, <span class="dt">drop =</span> <span class="ot">FALSE</span>], <span class="dt">X.nuis =</span> X[, <span class="op">-</span><span class="dv">2</span>, <span class="dt">drop =</span> <span class="ot">FALSE</span>],
                             <span class="dt">Y =</span> Y, <span class="dt">r =</span> num_sv, <span class="dt">adj.method =</span> <span class="st">"nc"</span>, <span class="dt">nc =</span> ctl)
ruv4_ruv   &lt;-<span class="st"> </span>ruv<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/ruv/topics/RUV4">RUV4</a></span>(<span class="dt">Y =</span> Y, <span class="dt">X =</span> X[, <span class="dv">2</span>, <span class="dt">drop =</span> <span class="ot">FALSE</span>], <span class="dt">ctl =</span> ctl, <span class="dt">k =</span> num_sv,
                        <span class="dt">Z =</span> X[, <span class="op">-</span><span class="dv">2</span>, <span class="dt">drop =</span> <span class="ot">FALSE</span>])</code></pre></div>
<p>The ruv package also implements the RUV2 method.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ruv2_ruv &lt;-<span class="st"> </span>ruv<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/ruv/topics/RUV2">RUV2</a></span>(<span class="dt">Y =</span> Y, <span class="dt">X =</span> X[, <span class="dv">2</span>, <span class="dt">drop =</span> <span class="ot">FALSE</span>], <span class="dt">ctl =</span> ctl, <span class="dt">k =</span> num_sv,
                      <span class="dt">Z =</span> X[, <span class="op">-</span><span class="dv">2</span>, <span class="dt">drop =</span> <span class="ot">FALSE</span>])</code></pre></div>
<p>A method that is both a version of RUV2 and a version of RUV4 is implemented in the vicar function <code>ruv3</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ruv3_vicar &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ruv3.html">ruv3</a></span>(<span class="dt">Y =</span> Y, <span class="dt">X =</span> X, <span class="dt">k =</span> num_sv, <span class="dt">ctl =</span> ctl, <span class="dt">cov_of_interest =</span> <span class="dv">2</span>)</code></pre></div>
<p>Finally, a Bayesian version of RUV is implemented in the vicar function <code>ruvb</code>. I run the Gibbs sampler for much fewer iterations than what you should do in practice.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ruvb_vicar &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ruvb.html">ruvb</a></span>(<span class="dt">Y =</span> Y, <span class="dt">X =</span> X, <span class="dt">ctl =</span> ctl, <span class="dt">k =</span> num_sv, <span class="dt">cov_of_interest =</span> <span class="dv">2</span>,
                   <span class="dt">fa_args =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">display_progress =</span> <span class="ot">FALSE</span>, <span class="dt">nsamp =</span> <span class="dv">1000</span>, <span class="dt">thin =</span> <span class="dv">5</span>))</code></pre></div>
</div>
<div id="analysis-without-controls" class="section level1">
<h1 class="hasAnchor">
<a href="#analysis-without-controls" class="anchor"></a>Analysis Without Controls</h1>
<p>When control genes are not present, there are still plenty of options. Two implementations in vicar are <code>mouthwash</code> and <code>backwash</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mout &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mouthwash.html">mouthwash</a></span>(<span class="dt">Y =</span> Y, <span class="dt">X =</span> X, <span class="dt">k =</span> num_sv, <span class="dt">cov_of_interest =</span> <span class="dv">2</span>, <span class="dt">include_intercept =</span> <span class="ot">FALSE</span>)</code></pre></div>
<pre><code>## Running mouthwash on 20 x 2 matrix X and 20 x 1000 matrix Y.
##  - Computing independent basis using QR decomposition.
##  - Computation took 0.009 seconds.
##  - Running additional preprocessing steps.
##  - Computation took 0.001 seconds.
##  - Running second step of mouthwash:
##     + Estimating model parameters using EM.
##     + Computation took 5.491 seconds.
##     + Generating adaptive shrinkage (ash) output.</code></pre>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre><code>##     + Computation took 0.166 seconds.
##  - Second step took 6.085 seconds.
##  - Estimating additional hidden confounders.
##  - Computation took 0.012 seconds.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bout &lt;-<span class="st"> </span><span class="kw"><a href="../reference/backwash.html">backwash</a></span>(<span class="dt">Y =</span> Y, <span class="dt">X =</span> X, <span class="dt">k =</span> num_sv, <span class="dt">cov_of_interest =</span> <span class="dv">2</span>, <span class="dt">include_intercept =</span> <span class="ot">FALSE</span>)</code></pre></div>
<pre><code>##  - Computing independent basis using QR decomposition.
##  - Computation took 0.008 seconds.
##  - Running additional preprocessing steps.
##  - Computation took 0.001 seconds.
##  - Running second step of backwash:
##     + Initializing parameters for EM algorithm.</code></pre>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre><code>##     + Computation took 0.406 seconds.
##     + Running one round of parameter updates.
##     + Computation took 0.029 seconds.
##     + Estimating model parameters using EM.
##     + Computation took 18.584 seconds.
##     + Generating posterior statistics.
##     + Computation took 0.006 seconds.
##  - Second step took 19.928 seconds.
##  - Generating final backwash outputs.
##  - Computation took 0.019 seconds.</code></pre>
<p>In terms of other packages, you can use the <code>sva</code> function in the sva package, the <code>cate</code> function in the cate package, or the <code>leapp</code> function in the leapp package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cate_cate   &lt;-<span class="st"> </span>cate<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/cate/topics/cate">cate.fit</a></span>(<span class="dt">X.primary =</span> X[, <span class="dv">2</span>, <span class="dt">drop =</span> <span class="ot">FALSE</span>], <span class="dt">X.nuis =</span> X[, <span class="op">-</span><span class="dv">2</span>, <span class="dt">drop =</span> <span class="ot">FALSE</span>],
                              <span class="dt">Y =</span> Y, <span class="dt">r =</span> num_sv, <span class="dt">adj.method =</span> <span class="st">"rr"</span>)
leapp_leapp &lt;-<span class="st"> </span>leapp<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/leapp/topics/leapp">leapp</a></span>(<span class="dt">data =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(Y), <span class="dt">pred.prim =</span> X[, <span class="dv">2</span>, <span class="dt">drop =</span> <span class="ot">FALSE</span>], 
                            <span class="dt">pred.covar =</span> X[, <span class="op">-</span><span class="dv">2</span>, <span class="dt">drop =</span> <span class="ot">FALSE</span>], <span class="dt">num.fac =</span> num_sv)

## Recommended pipeline for SVA
sva_sva     &lt;-<span class="st"> </span>sva<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/sva/topics/sva">sva</a></span>(<span class="dt">dat =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(Y), <span class="dt">mod =</span> X, <span class="dt">mod0 =</span> X[, <span class="op">-</span><span class="dv">2</span>, <span class="dt">drop =</span> <span class="ot">FALSE</span>], <span class="dt">n.sv =</span> num_sv)</code></pre></div>
<pre><code>## Number of significant surrogate variables is:  3 
## Iteration (out of 5 ):1  2  3  4  5</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">X.sva &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cbind">cbind</a></span>(X, sva_sva<span class="op">$</span>sv)
lmout &lt;-<span class="st"> </span>limma<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/limma/topics/lmFit">lmFit</a></span>(<span class="dt">object =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(Y), <span class="dt">design =</span> X.sva)
eout  &lt;-<span class="st"> </span>limma<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/limma/topics/ebayes-deprecated">ebayes</a></span>(lmout)</code></pre></div>
<pre><code>## Warning: 'limma::ebayes' is deprecated.
## Use 'eBayes' instead.
## See help("Deprecated")</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">svaout           &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()
svaout<span class="op">$</span>betahat   &lt;-<span class="st"> </span>lmout<span class="op">$</span>coefficients[, <span class="dv">2</span>]
svaout<span class="op">$</span>sebetahat &lt;-<span class="st"> </span>lmout<span class="op">$</span>stdev.unscaled[, <span class="dv">2</span>] <span class="op">*</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">sqrt</a></span>(eout<span class="op">$</span>s2.post)
svaout<span class="op">$</span>pvalues   &lt;-<span class="st"> </span>eout<span class="op">$</span>p.value[, <span class="dv">2</span>]
## Sanity check
## plot(lmout$coefficients[,2] / (lmout$stdev.unscaled[, 2] * sqrt(eout$s2.post)), eout$t[,2])</code></pre></div>
</div>
<div id="compare-methods" class="section level1">
<h1 class="hasAnchor">
<a href="#compare-methods" class="anchor"></a>Compare methods</h1>
<p>We’ll compare how all of the methods performed on these data using the <code>pROC</code> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">roc_out &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(
  pROC<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(<span class="dt">response =</span> which_null, <span class="dt">predictor =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(mout<span class="op">$</span>result<span class="op">$</span>lfdr)),
  pROC<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(<span class="dt">response =</span> which_null, <span class="dt">predictor =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(bout<span class="op">$</span>result<span class="op">$</span>lfdr)),
  pROC<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(<span class="dt">response =</span> which_null, <span class="dt">predictor =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(cate_cate<span class="op">$</span>beta.p.value)),
  pROC<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(<span class="dt">response =</span> which_null, <span class="dt">predictor =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(svaout<span class="op">$</span>pvalues)),
  pROC<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(<span class="dt">response =</span> which_null, <span class="dt">predictor =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(leapp_leapp<span class="op">$</span>p)),
  pROC<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(<span class="dt">response =</span> which_null, <span class="dt">predictor =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruv2_ruv<span class="op">$</span>p)),
  pROC<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(<span class="dt">response =</span> which_null, <span class="dt">predictor =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruv3_vicar<span class="op">$</span>pvalues_unadjusted)),
  pROC<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(<span class="dt">response =</span> which_null, <span class="dt">predictor =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruv4_vicar<span class="op">$</span>pvalues)),
  pROC<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(<span class="dt">response =</span> which_null, <span class="dt">predictor =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruv4_cate<span class="op">$</span>beta.p.value)),
  pROC<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(<span class="dt">response =</span> which_null, <span class="dt">predictor =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruv4_ruv<span class="op">$</span>p)),
  pROC<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/pROC/topics/roc">roc</a></span>(<span class="dt">response =</span> which_null, <span class="dt">predictor =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruvb_vicar<span class="op">$</span>lfsr2)))
name_vec &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"MOUTHWASH"</span>, <span class="st">"BACKWASH"</span>, <span class="st">"CATErr"</span>, <span class="st">"SVA"</span>, <span class="st">"LEAPP"</span>, <span class="st">"RUV2"</span>, <span class="st">"RUV3"</span>, <span class="st">"RUV4v"</span>, <span class="st">"RUV4c"</span>, <span class="st">"RUV4r"</span>, <span class="st">"RUVb"</span>)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(roc_out) &lt;-<span class="st"> </span>name_vec

sout &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(roc_out, <span class="cf">function</span>(x) { <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">TPR =</span> x<span class="op">$</span>sensitivities, <span class="dt">FPR =</span> <span class="dv">1</span> <span class="op">-</span><span class="st"> </span>x<span class="op">$</span>specificities)})
<span class="cf">for</span> (index <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(sout)) {
  sout[[index]]<span class="op">$</span>Method &lt;-<span class="st"> </span>name_vec[index]
}
longdat &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/do.call">do.call</a></span>(rbind, sout)</code></pre></div>
<p>We’ll first look at the ROC Curves for the methods that use control genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">shortdat &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(longdat, Method <span class="op">==</span><span class="st"> "RUV2"</span> <span class="op">|</span><span class="st"> </span>Method <span class="op">==</span><span class="st"> "RUV3"</span> <span class="op">|</span><span class="st"> </span>Method <span class="op">==</span><span class="st"> "RUV4v"</span> <span class="op">|</span>
<span class="st">                            </span>Method <span class="op">==</span><span class="st"> "RUV4c"</span> <span class="op">|</span><span class="st"> </span>Method <span class="op">==</span><span class="st"> "RUV4r"</span> <span class="op">|</span><span class="st"> </span>Method <span class="op">==</span><span class="st"> "RUVb"</span>)
<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="dt">data =</span> shortdat, <span class="dt">mapping =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x =</span> FPR, <span class="dt">y =</span> TPR, <span class="dt">col =</span> Method)) <span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_path</a></span>() <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>() <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"ROC Curves"</span>)</code></pre></div>
<p><img src="sample_analysis_files/figure-html/unnamed-chunk-10-1.png" width="672"></p>
<p>Eyeballing it, it seems that RUV3 and RUV2 perform the best here, though not with regards to the most significant genes. The <code>ruv</code> version of RUV4 appears to do much worse.</p>
<p>Now we’ll look at the ROC curves of methods that do not use control genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">shortdat &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(longdat, Method <span class="op">==</span><span class="st"> "MOUTHWASH"</span> <span class="op">|</span><span class="st"> </span>Method <span class="op">==</span><span class="st"> "BACKWASH"</span> <span class="op">|</span>
<span class="st">                            </span>Method <span class="op">==</span><span class="st"> "CATErr"</span> <span class="op">|</span><span class="st"> </span>Method <span class="op">==</span><span class="st"> "SVA"</span> <span class="op">|</span><span class="st"> </span>Method <span class="op">==</span><span class="st"> "LEAPP"</span>)
<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="dt">data =</span> shortdat, <span class="dt">mapping =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x =</span> FPR, <span class="dt">y =</span> TPR, <span class="dt">col =</span> Method)) <span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_path</a></span>() <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>() <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"ROC Curves"</span>)</code></pre></div>
<p><img src="sample_analysis_files/figure-html/unnamed-chunk-11-1.png" width="672"></p>
<p>Eyeballing it, it seems that MOUTHWASH and BACKWASH do the best over the largest length of the curve. Note that the BACKWASH curve is nearly completely covered by the MOUTHWASH curve.</p>
<p>We can calculate the areas under the curve (AUC) for each method</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">auc_vec &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">sapply</a></span>(roc_out, <span class="dt">FUN =</span> <span class="cf">function</span>(x) { x<span class="op">$</span>auc })
knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(auc_vec, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>), <span class="dt">col.names =</span> <span class="st">"AUC"</span>, <span class="dt">digits =</span> <span class="dv">3</span>)</code></pre></div>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="right">AUC</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>MOUTHWASH</td>
<td align="right">0.892</td>
</tr>
<tr class="even">
<td>BACKWASH</td>
<td align="right">0.891</td>
</tr>
<tr class="odd">
<td>RUV3</td>
<td align="right">0.885</td>
</tr>
<tr class="even">
<td>CATErr</td>
<td align="right">0.884</td>
</tr>
<tr class="odd">
<td>RUV2</td>
<td align="right">0.884</td>
</tr>
<tr class="even">
<td>LEAPP</td>
<td align="right">0.879</td>
</tr>
<tr class="odd">
<td>RUVb</td>
<td align="right">0.875</td>
</tr>
<tr class="even">
<td>SVA</td>
<td align="right">0.871</td>
</tr>
<tr class="odd">
<td>RUV4v</td>
<td align="right">0.859</td>
</tr>
<tr class="even">
<td>RUV4c</td>
<td align="right">0.856</td>
</tr>
<tr class="odd">
<td>RUV4r</td>
<td align="right">0.816</td>
</tr>
</tbody>
</table>
</div>
<div id="estimating-the-proportion-of-genes-that-are-null-" class="section level1">
<h1 class="hasAnchor">
<a href="#estimating-the-proportion-of-genes-that-are-null-" class="anchor"></a>Estimating the proportion of genes that are NULL.</h1>
<p>It is sometimes of interest to estimate the number of genes that show a signal. <code>mouthwash</code> and <code>backwash</code> already return these estimates. For other methods, we can use a summary statistic method (SSM) to get estimates of the proportion of genes that are null. Two useful SSM’s are implemented in the ashr and qvalue R packages. We’ll look at using the ashr package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">method_list &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()
method_list<span class="op">$</span>CATErr           &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()
method_list<span class="op">$</span>CATErr<span class="op">$</span>betahat   &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(cate_cate<span class="op">$</span>beta)
method_list<span class="op">$</span>CATErr<span class="op">$</span>sebetahat &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">sqrt</a></span>(cate_cate<span class="op">$</span>beta.cov.row <span class="op">*</span><span class="st"> </span>cate_cate<span class="op">$</span>beta.cov.col) <span class="op">/</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">sqrt</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(X)))</code></pre></div>
<pre><code>## Warning in cate_cate$beta.cov.row * cate_cate$beta.cov.col: Recycling array of length 1 in vector-array arithmetic is deprecated.
##   Use c() or as.vector() instead.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">method_list<span class="op">$</span>RUV2            &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()
method_list<span class="op">$</span>RUV2<span class="op">$</span>betahat    &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruv2_ruv<span class="op">$</span>betahat)
method_list<span class="op">$</span>RUV2<span class="op">$</span>sebetahat  &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">sqrt</a></span>(ruv2_ruv<span class="op">$</span>multiplier <span class="op">*</span><span class="st"> </span>ruv2_ruv<span class="op">$</span>sigma2))</code></pre></div>
<pre><code>## Warning in ruv2_ruv$multiplier * ruv2_ruv$sigma2: Recycling array of length 1 in array-vector arithmetic is deprecated.
##   Use c() or as.vector() instead.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">method_list<span class="op">$</span>RUV3            &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()
method_list<span class="op">$</span>RUV3<span class="op">$</span>betahat    &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruv3_vicar<span class="op">$</span>betahat)
method_list<span class="op">$</span>RUV3<span class="op">$</span>sebetahat  &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruv3_vicar<span class="op">$</span>sebetahat_unadjusted)

method_list<span class="op">$</span>RUV4r           &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()
method_list<span class="op">$</span>RUV4r<span class="op">$</span>betahat   &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruv4_ruv<span class="op">$</span>betahat)
method_list<span class="op">$</span>RUV4r<span class="op">$</span>sebetahat &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">sqrt</a></span>(ruv4_ruv<span class="op">$</span>multiplier <span class="op">*</span><span class="st"> </span>ruv4_ruv<span class="op">$</span>sigma2))</code></pre></div>
<pre><code>## Warning in ruv4_ruv$multiplier * ruv4_ruv$sigma2: Recycling array of length 1 in array-vector arithmetic is deprecated.
##   Use c() or as.vector() instead.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">method_list<span class="op">$</span>RUV4v           &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()
method_list<span class="op">$</span>RUV4v<span class="op">$</span>betahat   &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruv4_vicar<span class="op">$</span>betahat)
method_list<span class="op">$</span>RUV4v<span class="op">$</span>sebetahat &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruv4_vicar<span class="op">$</span>sebetahat_ols)

method_list<span class="op">$</span>RUV4c           &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()
method_list<span class="op">$</span>RUV4c<span class="op">$</span>betahat   &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruv4_cate<span class="op">$</span>beta)
method_list<span class="op">$</span>RUV4c<span class="op">$</span>sebetahat &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">sqrt</a></span>(ruv4_cate<span class="op">$</span>beta.cov.row <span class="op">*</span><span class="st"> </span>ruv4_cate<span class="op">$</span>beta.cov.col) <span class="op">/</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">sqrt</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(X)))</code></pre></div>
<pre><code>## Warning in ruv4_cate$beta.cov.row * ruv4_cate$beta.cov.col: Recycling array of length 1 in vector-array arithmetic is deprecated.
##   Use c() or as.vector() instead.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">method_list<span class="op">$</span>RUVb            &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()
method_list<span class="op">$</span>RUVb<span class="op">$</span>betahat    &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruvb_vicar<span class="op">$</span>means)
method_list<span class="op">$</span>RUVb<span class="op">$</span>sebetahat  &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(ruvb_vicar<span class="op">$</span>sd)

method_list<span class="op">$</span>SVA             &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()
method_list<span class="op">$</span>SVA<span class="op">$</span>betahat     &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(svaout<span class="op">$</span>betahat)
method_list<span class="op">$</span>SVA<span class="op">$</span>sebetahat   &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(svaout<span class="op">$</span>sebetahat)

ashfit &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(method_list, <span class="dt">FUN =</span> <span class="cf">function</span>(x) { ashr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/ashr/topics/ash">ash</a></span>(x<span class="op">$</span>betahat, x<span class="op">$</span>sebetahat)})</code></pre></div>
<pre><code>## Due to absence of package REBayes, switching to EM algorithm
## Due to absence of package REBayes, switching to EM algorithm
## Due to absence of package REBayes, switching to EM algorithm
## Due to absence of package REBayes, switching to EM algorithm
## Due to absence of package REBayes, switching to EM algorithm
## Due to absence of package REBayes, switching to EM algorithm
## Due to absence of package REBayes, switching to EM algorithm
## Due to absence of package REBayes, switching to EM algorithm</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">api0 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">sapply</a></span>(ashfit, <span class="dt">FUN =</span> ashr<span class="op">::</span>get_pi0)
api0 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(api0, <span class="dt">MOUTHWASH =</span> mout<span class="op">$</span>pi0)
api0 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(api0, <span class="dt">BACKWASH =</span> bout<span class="op">$</span>pi0)</code></pre></div>
<p>In these data, at least, MOUTHWASH, BACKWASH, and SVA have by far the most accurate estimates of the proportion of genes that are null (<span class="math inline">\(\pi_0\)</span>), which, recall, is 0.9.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sort">sort</a></span>(api0, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>), <span class="dt">col.names =</span> <span class="st">"Estimate of Pi0"</span>)</code></pre></div>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="right">Estimate of Pi0</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>RUV2</td>
<td align="right">0.9383859</td>
</tr>
<tr class="even">
<td>RUV3</td>
<td align="right">0.9351290</td>
</tr>
<tr class="odd">
<td>RUVb</td>
<td align="right">0.9215029</td>
</tr>
<tr class="even">
<td>SVA</td>
<td align="right">0.9072704</td>
</tr>
<tr class="odd">
<td>BACKWASH</td>
<td align="right">0.8968591</td>
</tr>
<tr class="even">
<td>MOUTHWASH</td>
<td align="right">0.8943147</td>
</tr>
<tr class="odd">
<td>RUV4v</td>
<td align="right">0.7292451</td>
</tr>
<tr class="even">
<td>CATErr</td>
<td align="right">0.7013801</td>
</tr>
<tr class="odd">
<td>RUV4c</td>
<td align="right">0.6990074</td>
</tr>
<tr class="even">
<td>RUV4r</td>
<td align="right">0.3923457</td>
</tr>
</tbody>
</table>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#abstract">Abstract</a></li>
      <li><a href="#data">Data</a></li>
      <li><a href="#analysis-with-controls">Analysis With Controls</a></li>
      <li><a href="#analysis-without-controls">Analysis Without Controls</a></li>
      <li><a href="#compare-methods">Compare methods</a></li>
      <li><a href="#estimating-the-proportion-of-genes-that-are-null-">Estimating the proportion of genes that are NULL.</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by David Gerard, Peter Carbonetto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.2.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
