<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Calibrated RUV4 where the control genes are used to estimate hidden
confounders and a variance inflation parameter. — vruv4 • vicar</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Calibrated RUV4 where the control genes are used to estimate hidden
confounders and a variance inflation parameter. — vruv4" />

<meta property="og:description" content="This function will perform a variant of Removing Unwanted Variation
4-step (RUV4) (Gagnon-Bartsch et al, 2013) where the control genes
are used not only to estimate the hidden confounders, but to
estimate a variance inflation parameter. This variance inflation
step is akin to the &quot;empirical null&quot; approach of Efron (2004)." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">vicar</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/customFA.html">Customizable Factor Analyses</a>
    </li>
    <li>
      <a href="../articles/custom_prior.html">Custom Prior for RUVB</a>
    </li>
    <li>
      <a href="../articles/sample_analysis.html">Sample Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/dcgerard/vicar">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calibrated RUV4 where the control genes are used to estimate hidden
confounders and a variance inflation parameter.</h1>
    <small class="dont-index">Source: <a href='https://github.com/dcgerard/vicar/blob/master/R/ruv4.R'><code>R/ruv4.R</code></a></small>
    <div class="hidden name"><code>vruv4.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function will perform a variant of Removing Unwanted Variation
4-step (RUV4) (Gagnon-Bartsch et al, 2013) where the control genes
are used not only to estimate the hidden confounders, but to
estimate a variance inflation parameter. This variance inflation
step is akin to the "empirical null" approach of Efron (2004).</p>
    
    </div>

    <pre class="usage"><span class='fu'>vruv4</span>(<span class='no'>Y</span>, <span class='no'>X</span>, <span class='no'>ctl</span>, <span class='kw'>k</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>cov_of_interest</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/nrow'>ncol</a></span>(<span class='no'>X</span>),
  <span class='kw'>likelihood</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"t"</span>, <span class='st'>"normal"</span>), <span class='kw'>limmashrink</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>degrees_freedom</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>include_intercept</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>gls</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>fa_func</span> <span class='kw'>=</span> <span class='no'>pca_naive</span>, <span class='kw'>fa_args</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/list'>list</a></span>(), <span class='kw'>adjust_bias</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>Y</th>
      <td><p>A matrix of numerics. These are the response variables
where each column has its own variance. In a gene expression
study, the rows are the individuals and the columns are the
genes.</p></td>
    </tr>
    <tr>
      <th>X</th>
      <td><p>A matrix of numerics. The covariates of interest.</p></td>
    </tr>
    <tr>
      <th>ctl</th>
      <td><p>A vector of logicals of length <code><a href='https://www.rdocumentation.org/packages/base/topics/nrow'>ncol(Y)</a></code>. If
position i is <code>TRUE</code> then position i is considered a
negative control.</p></td>
    </tr>
    <tr>
      <th>k</th>
      <td><p>A non-negative integer.The number of unobserved
confounders. If not specified and the R package sva is
installed, then this function will estimate the number of
hidden confounders using the methods of Buja and Eyuboglu
(1992).</p></td>
    </tr>
    <tr>
      <th>cov_of_interest</th>
      <td><p>A vector of positive integers. The column
numbers of the covariates in X whose coefficients you are
interested in. The rest are considered nuisance parameters and
are regressed out by OLS.</p></td>
    </tr>
    <tr>
      <th>likelihood</th>
      <td><p>Either <code>"normal"</code> or <code>"t"</code>. If
<code>likelihood = "t"</code>, then the user may provide the degrees
of freedom via <code>degrees_freedom</code>.</p></td>
    </tr>
    <tr>
      <th>limmashrink</th>
      <td><p>A logical. Should we apply hierarchical
shrinkage to the variances (<code>TRUE</code>) or not (<code>FALSE</code>)?
If <code>degrees_freedom = NULL</code> and <code>limmashrink = TRUE</code>
and <code>likelihood = "t"</code>, then we'll also use the limma
returned degrees of freedom.</p></td>
    </tr>
    <tr>
      <th>degrees_freedom</th>
      <td><p>if <code>likelihood = "t"</code>, then this is the
user-defined degrees of freedom for that distribution. If
<code>degrees_freedom</code> is <code>NULL</code> then the degrees of
freedom will be the sample size minus the number of covariates
minus <code>k</code>.</p></td>
    </tr>
    <tr>
      <th>include_intercept</th>
      <td><p>A logical. If <code>TRUE</code>, then it will
check <code>X</code> to see if it has an intercept term. If not, then
it will add an intercept term. If <code>FALSE</code>, then <code>X</code>
will be unchanged.</p></td>
    </tr>
    <tr>
      <th>gls</th>
      <td><p>A logical. Should we use generalized least squares
(<code>TRUE</code>) or ordinary least squares (<code>FALSE</code>) for
estimating the confounders? The OLS version is equivalent to
using RUV4 to estimate the confounders.</p></td>
    </tr>
    <tr>
      <th>fa_func</th>
      <td><p>A factor analysis function. The function must have
as inputs a numeric matrix <code>Y</code> and a rank (numeric scalar)
<code>r</code>. It must output numeric matrices <code>alpha</code> and
<code>Z</code> and a numeric vector <code>sig_diag</code>. <code>alpha</code> is
the estimate of the coefficients of the unobserved confounders,
so it must be an <code>r</code> by <code><a href='https://www.rdocumentation.org/packages/base/topics/nrow'>ncol(Y)</a></code> matrix. <code>Z</code>
must be an <code>r</code> by <code><a href='https://www.rdocumentation.org/packages/base/topics/nrow'>nrow(Y)</a></code> matrix. <code>sig_diag</code>
is the estimate of the column-wise variances so it must be of
length <code><a href='https://www.rdocumentation.org/packages/base/topics/nrow'>ncol(Y)</a></code>. The default is the function
<code>pca_naive</code> that just uses the first <code>r</code> singular
vectors as the estimate of <code>alpha</code>. The estimated
variances are just the column-wise mean square.</p></td>
    </tr>
    <tr>
      <th>fa_args</th>
      <td><p>A list. Additional arguments you want to pass to
fa_func.</p></td>
    </tr>
    <tr>
      <th>adjust_bias</th>
      <td><p>A logical. Should we also use the control genes
to adjust for bias (<code>TRUE</code>) or not (<code>FALSE</code>)</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list whose elements are:</p>
<p><code>multiplier</code> A numeric. The estimated variance inflation
    parameter.</p>
<p><code>betahat_ols</code> A vector of numerics. The ordinary least
    squares estimates of the coefficients of the covariate of
    interest. This is when not including the estimated confounding
    variables.</p>
<p><code>sebetahat_ols</code> A vector of positive numerics. The
    pre-inflation standard errors of <code>ruv$betahat</code> (NOT
    <code>ruv$betahat_ols</code>).</p>
<p><code>betahat</code> A matrix of numerics. The ordinary least squares
    estimates of the coefficients of the covariate of interest WHEN
    YOU ALSO INCLUDE THE ESTIMATES OF THE UNOBSERVED CONFOUNDERS.</p>
<p><code>sebetahat</code> A matrix of positive numerics. This is equal
    to <code>sebethat_ols * sqrt(multiplier)</code>. This is the
    post-inflation adjusted standard errors for <code>ruv$betahat</code>.</p>
<p><code>tstats</code> A vector of numerics. The t-statistics for
    testing against the null hypothesis of the coefficient of the
    covariate of interest being zero. This is after estimating the
    variance inflation parameter.</p>
<p><code>pvalues</code> A vector of numerics. The p-values of said test
    above.</p>
<p><code>alphahat</code> A matrix of numerics. The estimates of the
    coefficients of the hidden confounders. Only identified up to a
    rotation on the rowspace.</p>
<p><code>Zhat</code> A matrix of numerics. The estimates of the
    confounding variables. Only identified up to a rotation on the
    columnspace.</p>
<p><code>sigma2</code> A vector of positive numerics. The estimates of
    the variances PRIOR to inflation.</p>
<p><code>sigma2_adjusted</code> A vector of positive numerics. The
    estimates of the variances AFTER to inflation. This is equal to
    <code>sigma2 * multiplier</code>.</p>
<p><code>mult_matrix</code> A matrix of numerics. Equal to
    <code><a href='https://www.rdocumentation.org/packages/base/topics/solve'>solve(t(R22) %*% R22)</a></code>. One multiplies <code>sigma2</code> or
    <code>simga2_adjused</code> by the diagonal elements of
    <code>mult_matrix</code> to get the standard errors of
    <code>betahat</code>.</p>
<p><code>degrees_freedom</code> The degrees of freedom. If
    <code>likelihood = "t"</code>, then this was the degrees of freedom
    used.</p>
<p><code>R22</code> A matrix of numerics numeric. This is the submatrix
    of the R in the QR decomposition of X that corresponds to the
    covariates of interest. This is mostly returned for debugging
    purposes and may be removed in the future.</p>
<p><code>Z2</code> A matrix of numerics of length 1. This is the
    estimated confounders (after a rotation). Not useful on it's
    own and is mostly returned for debugging purposes. It may be
    removed in the future.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The model is $$Y = XB + ZA + E,$$ where \(Y\) is a matrix of
responses (e.g. log-transformed gene expression levels), \(X\) is
a matrix of covariates, \(B\) is a matrix of coefficients,
\(Z\) is a matrix of unobserved confounders, \(A\) is a matrix
of unobserved coefficients of the unobserved confounders, and
\(E\) is the noise matrix where the elements are independent
Gaussian and each column shares a common variance. The rows of
\(Y\) are the observations (e.g. individuals) and the columns of
\(Y\) are the response variables (e.g. genes).</p>
<p>This model is fit using a two-step approach proposed in
Gagnon-Bartsch et al (2013) and described in Wang et al (2015),
modified to include estimating a variance inflation parameter. An
additional modification is to use a t-likelihood in the second step
of the procedure, improving robustness to model misspecification.</p>
<p>For instructions and examples on how to specify your own factor analysis, run the following code in R:
<code><a href='../articles/customFA.html'>utils::vignette("customFA", package = "vicar")</a></code>. If it doesn't work, then you probably haven't built
the vignettes. To do so, see <a href='https://github.com/dcgerard/vicar#vignettes'>https://github.com/dcgerard/vicar#vignettes</a>.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Gagnon-Bartsch, J., Laurent Jacob, and Terence
    P. Speed. "Removing unwanted variation from high dimensional
    data with negative controls."
    Berkeley: Department of Statistics. University of California
    (2013).</p>
<p>Andreas Buja and Nermin
    Eyuboglu. "Remarks on parallel analysis." Multivariate behavior
    research, 27(4):509-540, 1992.</p>
<p>Bradley Efron
    "Large-Scale Simultaneous Hypothesis Testing: The Choice of a Null
    Hypothesis",
    Journal of the American Statistical Association, 99:465,
    96-104, 2004.</p>
<p>Wang, J., Zhao, Q., Hastie, T., &amp; Owen, A. B
    "Confounder Adjustment in Multiple Hypotheses Testing."
    arXiv preprint arXiv:1508.04178 (2015).</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='ruv3.html'>ruv3</a></code> for a version of RUV4 that can also be considered a version of RUV2.</p>
<p><code><a href='https://www.rdocumentation.org/packages/ruv/topics/RUV4'>RUV4</a></code> For the version of RUV4 in the ruv package.</p>
<p><code><a href='https://www.rdocumentation.org/packages/cate/topics/cate'>cate</a></code> For the version of RUV4 in the cate package.</p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>library</a></span>(<span class='no'>vicar</span>)

<span class='co'>## Generate data and controls ---------------------------------------------</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/Random'>set.seed</a></span>(<span class='fl'>1327</span>)
<span class='no'>n</span> <span class='kw'>&lt;-</span> <span class='fl'>13</span>
<span class='no'>p</span> <span class='kw'>&lt;-</span> <span class='fl'>201</span>
<span class='no'>k</span> <span class='kw'>&lt;-</span> <span class='fl'>2</span>
<span class='no'>q</span> <span class='kw'>&lt;-</span> <span class='fl'>3</span>
<span class='no'>is_null</span>       <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/rep'>rep</a></span>(<span class='fl'>FALSE</span>, <span class='kw'>length</span> <span class='kw'>=</span> <span class='no'>p</span>)
<span class='no'>is_null</span>[<span class='fl'>1</span>:<span class='fl'>101</span>] <span class='kw'>&lt;-</span> <span class='fl'>TRUE</span>
<span class='no'>ctl</span>           <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/rep'>rep</a></span>(<span class='fl'>FALSE</span>, <span class='kw'>length</span> <span class='kw'>=</span> <span class='no'>p</span>)
<span class='no'>ctl</span>[<span class='fl'>1</span>:<span class='fl'>73</span>]     <span class='kw'>&lt;-</span> <span class='fl'>TRUE</span>

<span class='no'>X</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/matrix'>matrix</a></span>(<span class='kw pkg'>stats</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/stats/topics/Normal'>rnorm</a></span>(<span class='no'>n</span> * <span class='no'>q</span>), <span class='kw'>nrow</span> <span class='kw'>=</span> <span class='no'>n</span>)
<span class='no'>B</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/matrix'>matrix</a></span>(<span class='kw pkg'>stats</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/stats/topics/Normal'>rnorm</a></span>(<span class='no'>q</span> * <span class='no'>p</span>), <span class='kw'>nrow</span> <span class='kw'>=</span> <span class='no'>q</span>)
<span class='no'>B</span>[<span class='fl'>2</span>, <span class='no'>is_null</span>] <span class='kw'>&lt;-</span> <span class='fl'>0</span>
<span class='no'>Z</span> <span class='kw'>&lt;-</span> <span class='no'>X</span> <span class='kw'>%*%</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/matrix'>matrix</a></span>(<span class='kw pkg'>stats</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/stats/topics/Normal'>rnorm</a></span>(<span class='no'>q</span> * <span class='no'>k</span>), <span class='kw'>nrow</span> <span class='kw'>=</span> <span class='no'>q</span>) +
     <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/matrix'>matrix</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/stats/topics/Normal'>rnorm</a></span>(<span class='no'>n</span> * <span class='no'>k</span>), <span class='kw'>nrow</span> <span class='kw'>=</span> <span class='no'>n</span>)
<span class='no'>A</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/matrix'>matrix</a></span>(<span class='kw pkg'>stats</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/stats/topics/Normal'>rnorm</a></span>(<span class='no'>k</span> * <span class='no'>p</span>), <span class='kw'>nrow</span> <span class='kw'>=</span> <span class='no'>k</span>)
<span class='no'>E</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/matrix'>matrix</a></span>(<span class='kw pkg'>stats</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/stats/topics/Normal'>rnorm</a></span>(<span class='no'>n</span> * <span class='no'>p</span>, <span class='kw'>sd</span> <span class='kw'>=</span> <span class='fl'>1</span> / <span class='fl'>2</span>), <span class='kw'>nrow</span> <span class='kw'>=</span> <span class='no'>n</span>)
<span class='no'>Y</span> <span class='kw'>&lt;-</span> <span class='no'>X</span> <span class='kw'>%*%</span> <span class='no'>B</span> + <span class='no'>Z</span> <span class='kw'>%*%</span> <span class='no'>A</span> + <span class='no'>E</span>

<span class='co'>## Fit RUV4 assuming only second covariate is of interest -----------------</span>
<span class='no'>ruvout</span> <span class='kw'>&lt;-</span> <span class='fu'>vruv4</span>(<span class='kw'>Y</span> <span class='kw'>=</span> <span class='no'>Y</span>, <span class='kw'>X</span> <span class='kw'>=</span> <span class='no'>X</span>, <span class='kw'>ctl</span> <span class='kw'>=</span> <span class='no'>ctl</span>, <span class='kw'>k</span> <span class='kw'>=</span> <span class='no'>k</span>, <span class='kw'>include_intercept</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
                <span class='kw'>likelihood</span> <span class='kw'>=</span> <span class='st'>"normal"</span>, <span class='kw'>cov_of_interest</span> <span class='kw'>=</span> <span class='fl'>2</span>)
<span class='kw pkg'>graphics</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/graphics/topics/plot'>plot</a></span>(<span class='no'>ruvout</span>$<span class='no'>pvalue</span>, <span class='kw'>col</span> <span class='kw'>=</span> <span class='no'>is_null</span> + <span class='fl'>3</span>, <span class='kw'>ylab</span> <span class='kw'>=</span> <span class='st'>"pvalues"</span>)</div><div class='input'><span class='kw pkg'>graphics</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/graphics/topics/legend'>legend</a></span>(<span class='st'>"topleft"</span>, <span class='kw'>col</span> <span class='kw'>=</span> <span class='fl'>3</span>:<span class='fl'>4</span>, <span class='kw'>legend</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"non-null"</span>, <span class='st'>"null"</span>),
                 <span class='kw'>pch</span> <span class='kw'>=</span> <span class='fl'>1</span>)</div><div class='img'><img src='vruv4-1.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'>## should look uniform</span>
<span class='kw pkg'>graphics</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/graphics/topics/hist'>hist</a></span>(<span class='no'>ruvout</span>$<span class='no'>pvalue</span>[<span class='no'>is_null</span>], <span class='kw'>xlab</span> <span class='kw'>=</span> <span class='st'>"pvalues"</span>, <span class='kw'>main</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</div><div class='img'><img src='vruv4-2.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Compare to linear model ------------------------------------------------</span>
<span class='no'>lmout</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/stats/topics/coef'>coefficients</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/summary'>summary</a></span>(<span class='kw pkg'>stats</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/stats/topics/lm'>lm</a></span>(<span class='no'>Y</span> ~ <span class='no'>X</span>)))
<span class='no'>lmp</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/lapply'>sapply</a></span>(<span class='no'>lmout</span>, <span class='kw'>function</span>(<span class='no'>x</span>) { <span class='no'>x</span>[<span class='fl'>3</span>, <span class='fl'>4</span>] })
<span class='kw pkg'>graphics</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/graphics/topics/plot'>plot</a></span>(<span class='no'>lmp</span>, <span class='kw'>col</span> <span class='kw'>=</span> <span class='no'>is_null</span> + <span class='fl'>3</span>, <span class='kw'>ylab</span> <span class='kw'>=</span> <span class='st'>"pvalues"</span>)</div><div class='input'><span class='kw pkg'>graphics</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/graphics/topics/legend'>legend</a></span>(<span class='st'>"topleft"</span>, <span class='kw'>col</span> <span class='kw'>=</span> <span class='fl'>3</span>:<span class='fl'>4</span>, <span class='kw'>legend</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"non-null"</span>, <span class='st'>"null"</span>),
                 <span class='kw'>pch</span> <span class='kw'>=</span> <span class='fl'>1</span>)</div><div class='img'><img src='vruv4-3.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'>## should look uniform</span>
<span class='kw pkg'>graphics</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/graphics/topics/hist'>hist</a></span>(<span class='no'>lmp</span>[<span class='no'>is_null</span>], <span class='kw'>xlab</span> <span class='kw'>=</span> <span class='st'>"pvalues"</span>, <span class='kw'>main</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</div><div class='img'><img src='vruv4-4.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Other ways to fit RUV4 from various packages ------------------------------</span>
<span class='no'>ruv4out</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>cate</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/cate/topics/cate'>cate.fit</a></span>(<span class='kw'>Y</span> <span class='kw'>=</span> <span class='no'>Y</span>, <span class='kw'>X.primary</span> <span class='kw'>=</span> <span class='no'>X</span>[, <span class='fl'>2</span>, <span class='kw'>drop</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>],
                          <span class='kw'>X.nuis</span> <span class='kw'>=</span> <span class='no'>X</span>[, -<span class='fl'>2</span>, <span class='kw'>drop</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>], <span class='kw'>r</span> <span class='kw'>=</span> <span class='no'>k</span>, <span class='kw'>fa.method</span> <span class='kw'>=</span> <span class='st'>"pc"</span>,
                          <span class='kw'>adj.method</span> <span class='kw'>=</span> <span class='st'>"nc"</span>, <span class='kw'>nc</span> <span class='kw'>=</span> <span class='no'>ctl</span>, <span class='kw'>calibrate</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
                          <span class='kw'>nc.var.correction</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='no'>vruv4out</span> <span class='kw'>&lt;-</span> <span class='fu'>vruv4</span>(<span class='kw'>Y</span> <span class='kw'>=</span> <span class='no'>Y</span>, <span class='kw'>X</span> <span class='kw'>=</span> <span class='no'>X</span>, <span class='kw'>k</span> <span class='kw'>=</span> <span class='no'>k</span>, <span class='kw'>cov_of_interest</span> <span class='kw'>=</span> <span class='fl'>2</span>,
                  <span class='kw'>include_intercept</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>ctl</span> <span class='kw'>=</span> <span class='no'>ctl</span>,
                  <span class='kw'>limmashrink</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>likelihood</span> <span class='kw'>=</span> <span class='st'>"normal"</span>)
<span class='no'>vruv4ols</span> <span class='kw'>&lt;-</span> <span class='fu'>vruv4</span>(<span class='kw'>Y</span> <span class='kw'>=</span> <span class='no'>Y</span>, <span class='kw'>X</span> <span class='kw'>=</span> <span class='no'>X</span>, <span class='kw'>k</span> <span class='kw'>=</span> <span class='no'>k</span>, <span class='kw'>cov_of_interest</span> <span class='kw'>=</span> <span class='fl'>2</span>,
                  <span class='kw'>include_intercept</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>ctl</span> <span class='kw'>=</span> <span class='no'>ctl</span>,
                  <span class='kw'>limmashrink</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>likelihood</span> <span class='kw'>=</span> <span class='st'>"normal"</span>,
                  <span class='kw'>gls</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='no'>ruv4ols</span>  <span class='kw'>&lt;-</span> <span class='kw pkg'>ruv</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/ruv/topics/RUV4'>RUV4</a></span>(<span class='kw'>Y</span> <span class='kw'>=</span> <span class='no'>Y</span>, <span class='kw'>X</span> <span class='kw'>=</span> <span class='no'>X</span>[, <span class='fl'>2</span>, <span class='kw'>drop</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>],
                      <span class='kw'>Z</span> <span class='kw'>=</span> <span class='no'>X</span>[, -<span class='fl'>2</span>, <span class='kw'>drop</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>], <span class='kw'>ctl</span> <span class='kw'>=</span> <span class='no'>ctl</span>, <span class='kw'>k</span> <span class='kw'>=</span> <span class='no'>k</span>)

<span class='no'>ruv4p</span> <span class='kw'>&lt;-</span> <span class='no'>ruv4out</span>$<span class='no'>beta.p.value</span>
<span class='no'>vruv4p</span> <span class='kw'>&lt;-</span> <span class='no'>vruv4out</span>$<span class='no'>pvalues</span>
<span class='no'>ruv4olsp</span> <span class='kw'>&lt;-</span> <span class='no'>ruv4ols</span>$<span class='no'>p</span>
<span class='no'>vruv4olsp</span> <span class='kw'>&lt;-</span> <span class='no'>vruv4ols</span>$<span class='no'>pvalues</span>

<span class='co'>## These should be monotonically related</span>
<span class='co'>## They often aren't because CATE often returns p-values that are exactly 0.</span>
<span class='kw pkg'>graphics</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/graphics/topics/plot'>plot</a></span>(<span class='no'>ruv4p</span>, <span class='no'>vruv4p</span>)</div><div class='img'><img src='vruv4-5.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/stats/topics/cor'>cor</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='no'>ruv4p</span>), <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='no'>vruv4p</span>), <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"kendall"</span>)</div><div class='output co'>#&gt; [1] 0.9883652</div><div class='input'>
<span class='co'>## These should be monotonically related</span>
<span class='kw pkg'>graphics</span><span class='kw ns'>::</span><span class='fu'><a href='https://www.rdocumentation.org/packages/graphics/topics/plot'>plot</a></span>(<span class='no'>ruv4olsp</span>, <span class='no'>vruv4olsp</span>)</div><div class='img'><img src='vruv4-6.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/stats/topics/cor'>cor</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='no'>ruv4olsp</span>), <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='no'>vruv4olsp</span>), <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"kendall"</span>)</div><div class='output co'>#&gt; [1] 1</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#references">References</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>David Gerard</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by David Gerard, Peter Carbonetto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.2.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

